service/history/workflow/task_generator.go

```diff
@@ -615,21 +615,37 @@ func (r *TaskGeneratorImpl) GenerateMigrationTasks() (tasks.Task, int64, error)
 		return nil, 0, err
 	}
 
+	workflowKey := r.mutableState.GetWorkflowKey()
+
 	if r.mutableState.GetExecutionState().State == enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
-		return &tasks.SyncWorkflowStateTask{
+		return []tasks.Task{&tasks.SyncWorkflowStateTask{
 			// TaskID, VisibilityTimestamp is set by shard
-			WorkflowKey: r.mutableState.GetWorkflowKey(),
+			WorkflowKey: workflowKey,
 			Version:     lastItem.GetVersion(),
-		}, 1, nil
-	} else {
-		return &tasks.HistoryReplicationTask{
-			// TaskID, VisibilityTimestamp is set by shard
-			WorkflowKey:  r.mutableState.GetWorkflowKey(),
-			FirstEventID: executionInfo.LastFirstEventId,
-			NextEventID:  lastItem.GetEventId() + 1,
-			Version:      lastItem.GetVersion(),
-		}, executionInfo.StateTransitionCount, nil
+		}}, 1, nil
 	}
+
+	now := time.Now().UTC()
+	replicationTasks := make([]tasks.Task, 0, len(r.mutableState.GetPendingActivityInfos())+1)
+	replicationTasks = append(replicationTasks, &tasks.HistoryReplicationTask{
```

Comment:
Q: Does the order of replicationTasks matters? i.e. active side is generating [history task, activity task], in the passive side, we are also expecting the cluster to process history task first, then process activity task?

Reply:
order should not matter on the sender sender side
on receiver side, i believe history tasks are sorted before activity tasks